google/cloud/storage/tests/testbench.py

```diff
@@ -693,6 +703,89 @@ def insert(self, gcs_url, request):
         self._insert_revision(revision)
         return revision
 
+    def _parse_part(self, multipart_upload_part):
+        """Parse a portion of a multipart breaking out the headers and payload.
+
+        :param multipart_upload_part:str a portion of the multipart upload body.
+        :return: a tuple with the headers and the payload.
+        :rtype: (dict, str)
+        """
+        headers = dict()
+        index = 0
+        next_line = multipart_upload_part.find('\r\n', index)
+        while next_line != index:
+            header_line = multipart_upload_part[index:next_line]
+            key, value = header_line.split(':', 2)
+            # This does not work for repeated headers, but we do not expect
+            # those in the testbench.
+            headers[key.encode('ascii', 'ignore')] = value
+            index = next_line + 2
+            next_line = multipart_upload_part.find('\r\n', index)
+        return headers, multipart_upload_part[next_line + 2:]
+
+    def insert_multipart(self, gcs_url, request):
+        """Insert a new revision based on the give flask request.
+
+        :param gcs_url:str the root URL for the fake GCS service.
+        :param request:flask.Request the contents of the HTTP request.
+        :return: the newly created object version.
+        :rtype: GcsObjectVersion
+        """
+        content_type = request.headers.get('content-type')
+        if content_type is None or not content_type.startswith('multipart/related'):
+            raise ErrorResponse('Missing or invalid content-type header in multipart upload')
+        _, _, boundary = content_type.partition('boundary=')
+        if boundary is None:
+            raise ErrorResponse('Missing boundary in content-type header in multipart upload')
```

Comment:
Do you think it's worth adding the expected boundary string to the error messages (here and on line 751)?

Reply:
Done.